src/openforms/registrations/contrib/objects_api/tests/test_api_endpoints.py

```diff
@@ -0,0 +1,176 @@
+from unittest.mock import patch
+
+import requests_mock
+from rest_framework import status
+from rest_framework.reverse import reverse_lazy
+from rest_framework.test import APITestCase
+from zgw_consumers.models import Service
+
+from openforms.accounts.tests.factories import StaffUserFactory, UserFactory
+
+from ..models import ObjectsAPIConfig
+
+
+@requests_mock.Mocker()
+class ObjecttypesAPIEndpointsTests(APITestCase):
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.endpoint = reverse_lazy("api:objects_api:object-types")
+
+        patcher = patch(
+            "openforms.registrations.contrib.objects_api.client.ObjectsAPIConfig.get_solo",
+            return_value=ObjectsAPIConfig(
+                objecttypes_service=Service(api_root="https://objecttypen.nl/api/v2/")
+            ),
+        )
```

Comment:
I don't like this. I'm touching implementation details of the client but this is not what I'm testing. A better solution would have been to create a mock client class, provide an mock implementation for the `get_objecttypes`/`get_objecttype_versions` methods, but I generally don't like going this way (considering the many footguns in the `unittest.mock` module. Having to rely on `mock` is also a sign that we are missing proper DI, however I don't know how this could be properly implemented in the drf ecosystem

Reply:
discussion moved to https://github.com/open-formulieren/open-forms/pull/3860#discussion_r1486262781